### PR TITLE
Fixed #22490 -- Added tests for Feed.get_object().

### PR DESCRIPTION
**Pull request by operi**
From this [ticket](https://code.djangoproject.com/ticket/22490). If there's any doubt or fix that needs to be made, please let me know.

### CODE REVIEW

File: tests/syndication_tests/tests.py [hunk 1]
```diff
@@ -522,3 +526,25 @@ def test_add_domain(self):
...
+        response = self.client.get('/syndication/rss2/%s/' % self.e1.pk)
+        doc = minidom.parseString(response.content)
+        feed = doc.getElementsByTagName('rss')[0]
+        chan = feed.getElementsByTagName('channel')[0]
+        items = chan.getElementsByTagName('item')
+
+        self.assertChildNodeContent(items[0], {
+            'title': 'Title: My first article',
+            'description': 'Article description: My first article',
+        })
```
Comment (felixxm) [inline tests/syndication_tests/tests.py:543]: I added assertions for `link`, `comments`, and `pubDate`.

**Review by felixxm — COMMENTED** [review]
@operi Thanks :+1: Welcome aboard :boat: 

I pushed minor edits.

**Comment by operi** [comment]
Awesome @felixxm ! Thank you!

File: tests/syndication_tests/tests.py [hunk 2]
```diff
@@ -522,3 +527,22 @@ def test_add_domain(self):
...
+
+    def test_get_object(self):
+        response = self.client.get('/syndication/rss2/articles/%s/' % self.e1.pk)
+        doc = minidom.parseString(response.content)
+        feed = doc.getElementsByTagName('rss')[0]
+        chan = feed.getElementsByTagName('channel')[0]
+        items = chan.getElementsByTagName('item')
+
+        self.assertChildNodeContent(items[0], {
+            'comments': '/blog/1/article/1/comments',
```
Comment (timgraham) [inline tests/syndication_tests/tests.py:539]: Is it possible not to hardcode the the primary keys to account for databases that use non-serial pks? Actually there are number of syndication tests that have the issue (observed on CockroachDB):
```
======================================================================
ERROR: test_rss2_feed (syndication_tests.tests.SyndicationFeedTest)
Test the structure and content of feeds generated by Rss201rev2Feed.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tim/code/django/tests/syndication_tests/tests.py", line 132, in test_rss2_feed
    d = Entry.objects.get(pk=1).published
  File "/home/tim/code/django/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/home/tim/code/django/django/db/models/query.py", line 435, in get
    raise self.model.DoesNotExist(
syndication_tests.models.Entry.DoesNotExist: Entry matching query does not exist.

======================================================================
FAIL: test_get_object (syndication_tests.tests.SyndicationFeedTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tim/code/django/tests/syndication_tests/tests.py", line 538, in test_get_object
    self.assertChildNodeContent(items[0], {
  File "/home/tim/code/django/tests/syndication_tests/tests.py", line 55, in assertChildNodeContent
    self.assertEqual(
AssertionError: '/blog/599055542775578625/article/599055542841442305/comments' != '/blog/1/article/1/comments'
- /blog/599055542775578625/article/599055542841442305/comments
+ /blog/1/article/1/comments


======================================================================
FAIL: test_latest_post_date (syndication_tests.tests.SyndicationFeedTest)
Both the published and updated dates are
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tim/code/django/tests/syndication_tests/tests.py", line 350, in test_latest_post_date
    self.assertEqual(updated, latest_updated)
AssertionError: '2013-03-25T20:00:00-05:00' != '2013-01-20T00:00:00-06:00'
- 2013-03-25T20:00:00-05:00
?       ^  ^ ^         ^
+ 2013-01-20T00:00:00-06:00
?       ^  ^ ^         ^


======================================================================
FAIL: test_rss091_feed (syndication_tests.tests.SyndicationFeedTest)
Test the structure and content of feeds generated by RssUserland091Feed.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tim/code/django/tests/syndication_tests/tests.py", line 252, in test_rss091_feed
    self.assertChildNodeContent(items[0], {
  File "/home/tim/code/django/tests/syndication_tests/tests.py", line 55, in assertChildNodeContent
    self.assertEqual(
AssertionError: 'http://example.com/blog/599055542775578625/' != 'http://example.com/blog/1/'
- http://example.com/blog/599055542775578625/
+ http://example.com/blog/1/


======================================================================
FAIL: test_template_feed (syndication_tests.tests.SyndicationFeedTest)
The item title and description can be overridden with templates.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tim/code/django/tests/syndication_tests/tests.py", line 493, in test_template_feed
    self.assertChildNodeContent(items[0], {
  File "/home/tim/code/django/tests/syndication_tests/tests.py", line 55, in assertChildNodeContent
    self.assertEqual(
AssertionError: 'http://example.com/blog/599055542775578625/' != 'http://example.com/blog/1/'
- http://example.com/blog/599055542775578625/
+ http://example.com/blog/1/
```

Comment (operi) [reply tests/syndication_tests/tests.py:539]: @timgraham Sure, I believe we could do something like` '/blog/%s/article/%s/comments' % self.e1.pk, self.a1.pk` (not sure if syntax is correct), right?

I can do those fixes in the failing tests. But I'm not sure of the proper way of handling it. What I mean is, should I create an issue on Trac and assign it to me?

Comment (timgraham) [reply tests/syndication_tests/tests.py:539]: You can make the changes without a ticket. See https://github.com/django/django/pull/12078 for a precedent. Thanks!

**Pull request merged**